test/Streamly/Test/Data/Parser/ParserD.hs

```diff
@@ -162,11 +176,14 @@ takeBetweenPass =
         forAll (chooseInt (m, max_value)) $ \n ->
             forAll (chooseInt (m, max_value)) $ \list_length ->
                 forAll (vectorOf list_length (chooseInt (min_value, max_value))) $ \ls ->
-                    case S.parseD (P.takeBetween m n FL.toList) (S.fromList ls) of
-                        Right parsed_list ->
-                            let lpl = Prelude.length parsed_list
-                            in checkListEqual parsed_list (Prelude.take lpl ls)
-                        Left _ -> property False
+                    monadicIO $ do
```

Comment:
Could use `runIdentity` everywhere instead of converting the code to IO monad. monadicIO is not required in most (or any) places in this file. it would be simpler to convert code to use `runIdentity`, that would be just one word change.